### ISSUE TITLE
"Report Anonymously" button background color becomes invisible after typing all form information

### ISSUE THREAD

**Issue by softvision-oana-arbuzov** [webcompat/webcompat.com#2151]
URL: https://staging.webcompat.com 
Browser / Version: Firefox Nightly 60.0a1 (2018-02-18) (Mobile)
Operating System: Google Pixel (Android 8.0.0) - 1080 x 1920 pixels (~441 ppi pixel density)

**Steps to Reproduce:**
    1. Navigate to https://staging.webcompat.com
    2. Tap "Report bug" button.
    3. Type in all required information.
    4. Observe "Report Anonymously" button.

**Expected Behavior:**
    Button has background color (gray).

**Actual Behavior:**
     Button has no background color.

**Note**
    1. Not reproducible on Desktop.
    2. Not reproducible on Chrome 64.0.3282.137 (Mobile).
    3. Not reproducible when hovering over the button. 
    4. Affected area:
```html
<button class="button js-Button" id="submitanon" name="submit-type" type="submit" value="github-proxy-report">
        Report Anonymously
</button>
``` 
        
**Watchers:**
@softvision-sergiulogigan
@softvision-oana-arbuzov

![screenshot-1519034581616](https://user-images.githubusercontent.com/12184325/36372153-d42ec42c-156c-11e8-8ad2-8b015cb48e7b.jpg)

**Comment by adamopenweb** [comment]
This button is displaying as grey for me. Maybe it's fixed?

**Comment by softvision-oana-arbuzov** [comment]
It is still reproducible for me.

**Tested with:**
Browser / Version: Firefox Mobile Nightly 61.0a1 (2018-03-15)
Operating System: Samsung Galaxy S7 Edge (Android 7.0) - Resolution 1440 x 2560 pixels (~534 ppi pixel density)

**Comment by miketaylr** [comment]
@softvision-oana-arbuzov can you post a screenshot? It WFM.

Chrome:

![chrome](https://user-images.githubusercontent.com/67283/37547984-e0f8ddf2-2942-11e8-944e-ea302e676802.png)

Firefox Nightly:
![screenshot_20180316-175136](https://user-images.githubusercontent.com/67283/37547990-e72f22a8-2942-11e8-82b3-3e011fd583da.png)

**Comment by softvision-oana-arbuzov** [comment]
@miketaylr, the highlight become hidden only after typing the required information.
![highlightmissing](https://user-images.githubusercontent.com/12184325/37584856-b806a414-2b5f-11e8-8414-c717b58d405b.gif)

**Tested with:**
Browser / Version: Firefox Nightly 61.0a1 (2018-03-18) (Mobile)
Operating System: Google Pixel (Android 8.0.0) - 1080 x 1920 pixels (~441 ppi pixel density)

**Comment by zoepage** [comment]
![screenshot_20180319-093304](https://user-images.githubusercontent.com/562354/37585226-ad37b69c-2b58-11e8-8f2c-064b4360c8c1.png)

After filling out the information, I can reproduce.

**Comment by miketaylr** [comment]
Ah, thanks. I missed that part. Let me take this bug.

**Comment by miketaylr** [comment]
Here's why this doesn't reproduce on Desktop, but does on mobile:

The `.button` class has `background-color: inherit`, which for desktop results in something gray ~ `color: ButtonText;`

https://searchfox.org/mozilla-central/source/toolkit/themes/osx/global/button.css#13

Fennec doesn't have a background color style defined for a `button` (which is weird, imho):

https://searchfox.org/mozilla-central/source/mobile/android/themes/geckoview/content.css#167

But when it's disabled (because the form is not valid), it gets the default global `color: GrayText` for disabled buttons here:

https://searchfox.org/mozilla-central/source/layout/style/res/forms.css#788

So it's doing what we told it to, just not doing what we expected. 

I think we're just missing a `button-secondary` class here, because it's defined but never used.

**Comment by miketaylr** [comment]
Oh wait, I put the color values, not the background-color values... let me update those links.

**Comment by zoepage** [comment]
Wow, that is a really good find. Thanks @miketaylr :)))

**Comment by zoepage** [comment]
merged #2261

**Comment by karlcow** [comment]
@miketaylr 

> Fennec doesn't have a background color style defined for a button (which is weird, imho):

Agreed about weird. This is https://bugzilla.mozilla.org/show_bug.cgi?id=1435665
